### PR TITLE
Add npm nodejsproj local deps

### PR DESCRIPTION
**Pull request by avinashkurup**
Hi Shahid,
    This PR adds NodeJS NPM settings, Python settings and Git hooks to Node and Python projects.

### CODE REVIEW

File: node-settings.ts [hunk 1]
```diff
@@ -10,35 +10,104 @@ export const nodeExtensions: Extension[] = [
...
+      test: "alsatian './**/*.spec.ts'",
+      lint: "eslint . --ext .ts",
+    },
+    repository: config.repository || {
+      url: "<no repository provided>",
+    },
+    publishConfig: config.publishConfig || {
+      registry: "https://npm.pkg.github.com/",
+    },
+    keywords: [
```
Comment (shah) [inline node-settings.ts]: These keywords should look like `keywords: config.keywords || []` (empty by default).

File: node-settings.ts [hunk 2]
```diff
@@ -10,35 +10,104 @@ export const nodeExtensions: Extension[] = [
...
+      "url",
+      "uri",
+      "urn",
+    ],
+    author: config.author || "<no author provided>",
+    license: config.license || "MIT",
+    dependencies: config.dependencies || {
+      "npm-package-name": "npm-package-version",
+    },
+    bugs: {
```
Comment (shah) [inline node-settings.ts]: This should be empty by default but allowed to be overridden like `bugs: config.bugs || {}`

File: project.ts [hunk 3]
```diff
@@ -202,31 +199,13 @@ export function enrichGitWorkTree(
...
 
   if (fs.existsSync(gitTreePath)) {
     const result: GitWorkTree = {
       ...pp,
       isGitWorkTree: true,
       gitDir: gitTreePath,
       gitWorkTree: workingTreePath,
-      gitConfig: {
-        preCommitHookFileName: gitCheckFileName,
-        writeSettings: (settings: gitSettings.GitCommitCheckSettings): void => {
```
Comment (shah) [inline project.ts:215]: The `writeSettings` method is good to keep here - meaning when we need to write the settings, you use this method but the actual content that is written is given from elsewhere.

File: project.ts [hunk 4]
```diff
@@ -600,6 +611,17 @@ export function enrichNodeProject(
           );
         }
       },
+      writeGitPrecommitHook: (
+        gitPrecommitCmd: gitSettings.GitCommitCheckDefn,
+      ) => {
+        if (fs.existsSync(gitTreePath)) {
+          Deno.writeTextFileSync(
```
Comment (shah) [inline project.ts]: The actual writing should be done by the git settings methods (see previous comment). Calling `git.writeSettings()` is good here, but the actual writing is done in the git settings.

**Review by shah — COMMENTED** [review]
Please fix these and then let's do a full walk through again before pushing.

File: project.ts (hunk 4 above)
Comment (avinashkurup) [reply project.ts]: Hi Shahid,
    I have made all the above changes and pushed them to this branch. Please review them.

File: project.ts (hunk 3 above)
Comment (avinashkurup) [reply project.ts:215]: Hi Shahid,
    Please check the latest changes as we discussed in today's code review. It adds generalizations for Linux executables used for creating Git pre-commit hooks.

File: project.ts (hunk 4 above)
Comment (avinashkurup) [reply project.ts]: Hi Shahid,
    Please check the latest changes as we had discussed in today's code review. It adds a generalization for Linux executables used for creating Git pre-commit checks.

**Pull request merged**